zjit/src/hir.rs

```diff
@@ -2594,10 +2606,10 @@ mod infer_tests {
     fn test_unknown() {
         crate::cruby::with_rubyvm(|| {
             let mut function = Function::new(std::ptr::null());
-            let param = function.push_insn(function.entry_block, Insn::PutSelf);
+            let param = function.push_insn(function.entry_block, Insn::Param { idx: SELF_PARAM_IDX });
             let val = function.push_insn(function.entry_block, Insn::Test { val: param });
             function.infer_types();
-            assert_bit_equal(function.type_of(val), types::CBool);
+            assert_bit_equal(function.type_of(val), Type::from_cbool(false));
```

Comment:
Why did this change? We shouldn't know that it's `false`

Reply:
Aha, we need to do

```rust
            function.param_types.push(types::BasicObject); // self
```

Reply:
(in the test)

Reply:
Thanks for pointing out the fix. Yeah I was looking into this and it had `Empty` and `is_known_falsy()` returned true (?). Pushed the fix at https://github.com/ruby/ruby/pull/13529/commits/5a543a65b08987fc5539e5399cab423f785de48b.

I wish we had less synthetic tests like this. It seems to test something unreal and requires non-trivial effort for debugging it. The more integrated, the easier to maintain.